kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/convert/BigQueryRecordConverter.java

```diff
@@ -73,6 +73,14 @@ public BigQueryRecordConverter(boolean shouldConvertDoubleSpecial) {
    */
   public Map<String, Object> convertRecord(SinkRecord kafkaConnectRecord) {
     Schema kafkaConnectSchema = kafkaConnectRecord.valueSchema();
+    Object kafkaConnectValue = kafkaConnectRecord.value();
+    if (kafkaConnectSchema == null) {
+      if (kafkaConnectValue instanceof Map) {
+        return (Map) kafkaConnectRecord;
```

Comment:
What if the map has non-string keys?
What about the `shouldDonvertSpecialDouble` config? Doesn't look like that's respected here.
What about the special conversion logic for bytes (currently at line 113)? Doesn't look like that's used here.
And, for all of the above, what about recursive descent with nested structures (e.g., if you have an array of `ByteBuffer`s, shouldn't any special conversion logic also be applied to that)?

Reply:
In addition, can some tests be written for this (especially if more fleshed-out logic is added)?
Has this been tested against an actual BigQuery instance at all? If so, with what data (or at least, what format of data)?

Reply:
maybe we can add explicit support for a defined set of basic data-types to start with, e.g. Map, List, String, Short, Long, Integer, Float, Double, and Boolean to start with, and make sure we handle those exactly the same way we do in the struct case, and throw a conversion exception in any other case.

This should cover all of the cases people would encounter when dealing with json data, which is the goal here.

Reply:
That sounds reasonable, the [JsonConverter](https://github.com/apache/kafka/blob/ff9e95cb09907739c17b4f4681b11c525515b995/connect/json/src/main/java/org/apache/kafka/connect/json/JsonConverter.java#L65-L186) code should provide a good list of the types you should expect to encounter.

Reply:
Sounds good! The code works with the metrics data which is simple. I'll add more logics to deal with potentially incompatible data type.